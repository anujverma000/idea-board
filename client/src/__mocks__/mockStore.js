import configureStore from 'redux-mock-store'
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState= {
    idea: {
        list: []
    }
}
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
store.replaceReducer(rootReducer);

export default store;