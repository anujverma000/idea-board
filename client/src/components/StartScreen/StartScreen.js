import React from "react";
import styled from "styled-components";
import welcome from './welcome.png';
import AddIdeaAction from "../NavBar/AddIdeaAction";

const StartScreen = () => {
  return(
    <Wrapper>
      <Welcome src={welcome} alt="Welcome" />
      <HelpTextTitle>More Than An Idea Board</HelpTextTitle>
      <HelpTextDesc>
        Clear visible mammer to brainstorm, create plans, manage projects and more.
      </HelpTextDesc>
      <AddIdeaAction/>
      <HelpListTitle>
        Highlighted Feature List
      </HelpListTitle>
      <HelpList>
        <HelpLitItem>Instant Add/Edit</HelpLitItem>
        <HelpLitItem>Update Notifications</HelpLitItem>
        <HelpLitItem>Permanent storage</HelpLitItem>
        <HelpLitItem>Offline availablity</HelpLitItem>
        <HelpLitItem>Multi facet sorting</HelpLitItem>
        <HelpLitItem>Mobile Friendly</HelpLitItem>
      </HelpList>
    </Wrapper>
  )
}


/**
 *  STYLES
 */
const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 460px;
  max-width: 100%;
  margin: 0 auto;
  padding: 0 24px;
  box-sizing: border-box;
  text-align: center;
`;

const Welcome = styled.img`
  width: 140px;
  height: 140px;
  margin: 42px auto 0
`;

const HelpTextTitle = styled.h2`
  color: ${props => props.theme.textColor};
  font-size: 24px;
  font-weight: bold;
`
const HelpTextDesc = styled.p`
  color: ${props => props.theme.textColor};
  font-size: 20px;
  font-weight: normal;
`
const HelpList = styled.ul`
  text-align: initial;
  padding-left: 25%;
  color: ${props => props.theme.textColorLight};
  margin-bottom: 32px;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  margin-left: 32px;
`
const HelpLitItem = styled.li`
  font-size: 16px;
  padding: 16px;
  background: #fff;
  margin: 12px 0;
  width: 40%;
`
const HelpListTitle = styled.div`
  margin-top: 50px;
  text-align: center;
  font-weight: bold;
`;
export default StartScreen;