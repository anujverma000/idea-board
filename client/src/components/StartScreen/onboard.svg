<svg height="200px" viewBox="0 0 230 170">
<g id="surface1">
<path style=" stroke:none;fill-rule:nonzero;fill:#333399;fill-opacity:1;" d="M 93.148438 31.730469 L 104.484375 31.730469 L 104.484375 102.390625 L 93.148438 102.390625 Z M 93.148438 31.730469 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:#6666CC;fill-opacity:1;" d="M 98.816406 167.789062 C 62.867188 167.789062 33.625 138.542969 33.625 102.59375 C 33.625 81.492188 43.941406 61.597656 61.226562 49.375 L 67.773438 58.632812 C 53.488281 68.734375 44.960938 85.171875 44.960938 102.59375 C 44.960938 132.292969 69.121094 156.449219 98.8125 156.449219 C 128.507812 156.449219 152.664062 132.292969 152.664062 102.59375 C 152.664062 85.171875 144.140625 68.734375 129.859375 58.632812 L 136.402344 49.375 C 153.6875 61.601562 164.007812 81.496094 164.007812 102.59375 C 164.007812 138.542969 134.765625 167.789062 98.816406 167.789062 Z M 98.816406 167.789062 "/>
</g>
</svg>
