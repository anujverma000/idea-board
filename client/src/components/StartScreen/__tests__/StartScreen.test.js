import React from 'react';
import renderer from 'react-test-renderer';
import StartScreen from '../StartScreen'
import mockStore from "../../../__mocks__/mockStore";
import { Provider } from "react-redux";

it('renders correctly', () => {
  const tree = renderer.create(
      <Provider store={mockStore}><StartScreen /></Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
