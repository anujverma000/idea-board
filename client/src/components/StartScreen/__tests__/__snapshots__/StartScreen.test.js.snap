// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bwzfXH fNQkPj"
>
  <img
    alt="Welcome"
    className="sc-htpNat jBzAON"
    src="welcome.png"
  />
  <h2
    className="sc-bxivhb bxbhqb"
  >
    More Than An Idea Board
  </h2>
  <p
    className="sc-ifAKCX bbnWBz"
  >
    Clear visible mammer to brainstorm, create plans, manage projects and more.
  </p>
  <button
    className="sc-bdVaJa krSCzb"
    onClick={[Function]}
  >
    Add new idea
  </button>
  <div
    className="sc-gzVnrw dBkTsN"
  >
    Highlighted Feature List
  </div>
  <ul
    className="sc-EHOje cWxXhy"
  >
    <li
      className="sc-bZQynM gTilDZ"
    >
      Instant Add/Edit
    </li>
    <li
      className="sc-bZQynM gTilDZ"
    >
      Update Notifications
    </li>
    <li
      className="sc-bZQynM gTilDZ"
    >
      Permanent storage
    </li>
    <li
      className="sc-bZQynM gTilDZ"
    >
      Offline availablity
    </li>
    <li
      className="sc-bZQynM gTilDZ"
    >
      Multi facet sorting
    </li>
    <li
      className="sc-bZQynM gTilDZ"
    >
      Mobile Friendly
    </li>
  </ul>
</div>
`;
